cli/compose/loader/loader.go

```diff
@@ -486,9 +486,17 @@ func LoadSecrets(source map[string]interface{}, workingDir string) (map[string]t
 		return secrets, err
 	}
 	for name, secret := range secrets {
-		if secret.External.External && secret.External.Name == "" {
-			secret.External.Name = name
-			secrets[name] = secret
+		if secret.External.External {
+			if secret.External.Name == "" {
+				secret.External.Name = name
+				secrets[name] = secret
+			} else {
+				logrus.Warnf("secret %s: secret.external.name is deprecated in favor of secret.name", name)
```

Comment:
There's an issue open for this warning with volumes #608

This has the same problem, we'll need to pass in a version here so that we only warn `if version >= 3.5`